src/docker/README.md

```diff
@@ -1,26 +1,58 @@
+        `/Users/yourusername/srv/docker/gitlab/logs:/var/log/gitlab`
+        `/Users/yourusername/srv/docker/gitlab/data:/var/opt/gitlab`
+    3. Change the jenkins volumes to 
+        `/Users/yourusername/srv/docker/jenkins:/var/jenkins_home`
+    4. In your Docker-Desktop go to `Settings > General > Choose file sharing   implementation for your containers` and switch to osxfs(Legacy). osxfs(Legacy) utilizes more resources of the system so make sure that the assigned resources are sufficient by going to `Settings > Resources` otherwise Docker-Desktop may go to forever start mode on Restarting.
+    5. Add `shm_size: '5gb'`under gitlab services.
+
+Like the instructions for Linux, for macOS users to start the containers, run `docker-compose up -d` from the `docker` folder. If you have any problems accessing the services in the containers, run `docker-compose up` by itself to see output from the services as they start.
 
-To start the containers, run `docker-compose up -d` from the `docker` folder. If you have problems accessing the services in the containers, run `docker-compose up` by itself to see output from the services as they start.
 
 ## Access GitLab
 
-To access `GitLab`, point your browser to `http://localhost:10080` and log in with `root` as the username and `password` as the password. Then create a user for Jenkins, impersonate that user, get its API key, set up test repos, etc. When creating webhooks to trigger Jenkins jobs, use `http://jenkins:8080` as the base URL.
+To access `GitLab`, you first need to create a user - `root` with some password. To do so, follow the following steps :
+    1. In the Gitlab containers terminal inside Docker Desktop, type `gitlab-rails console` and wait for at least a few minutes for the console to start. 
+    2. Once the console is started successfully, run the following commands in sequence at the console, noting that there are certain security rules to the password choice:
+        a. `user = User.new(username: 'root', email: 'root@root.com', name: 'root', password: 'setyourown', password_confirmation: 'setyourown')`
+        b. `user.skip_confirmation!` 
+        c. `user.save!`
+    3. Now, point your browser to `http://localhost:port` and log in with `root` as the username and `setyourown` as the password. Then create a user for Jenkins, impersonate that user, get its API key, set up test repos, etc. When creating webhooks to trigger Jenkins jobs, use `http://jenkins:8080` as the base URL.
+
 
-If you have trouble cloning a GitLab repository, it may be because you have a leftover host key from an SSH connection to a previous installation of GitLab in Docker. To troubleshoot, run `ssh -vT git@localhost -p 10022`.
+If you have trouble cloning a GitLab repository, it may be because you have a leftover host key from an SSH connection to a previous installation of GitLab in Docker. To troubleshoot, run `ssh -vT git@localhost -p port`.
```

Comment:
```suggestion
If you have trouble cloning a GitLab repository, it may be because you have a leftover host key from an SSH connection to a previous installation of GitLab in Docker. To troubleshoot, run `ssh -vT git@localhost -p 55522`.
```